File: src/test/java/TextFormatterTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class TextFormatterTest {

    @Test
    public void testUppercaseStrategy() {
        TextFormatterStrategy uppercaseStrategy = new UppercaseStrategy();
        assertEquals("HELLO WORLD", uppercaseStrategy.format("hello world"));
        assertEquals("JAVA PROGRAMMING", uppercaseStrategy.format("Java Programming"));
    }

    @Test
    public void testLowercaseStrategy() {
        TextFormatterStrategy lowercaseStrategy = new LowercaseStrategy();
        assertEquals("hello world", lowercaseStrategy.format("HELLO WORLD"));
        assertEquals("java programming", lowercaseStrategy.format("Java Programming"));
    }

    @Test
    public void testTitleCaseStrategy() {
        TextFormatterStrategy titleCaseStrategy = new TitleCaseStrategy();
        assertEquals("Hello World", titleCaseStrategy.format("hello world"));
        assertEquals("Java Programming", titleCaseStrategy.format("JAVA PROGRAMMING"));
    }

    @Test
    public void testTextFormatterContext() {
        TextFormatterContext context = new TextFormatterContext(new UppercaseStrategy());
        assertEquals("HELLO WORLD", context.format("hello world"));

        context.setStrategy(new LowercaseStrategy());
        assertEquals("hello world", context.format("HELLO WORLD"));

        context.setStrategy(new TitleCaseStrategy());
        assertEquals("Hello World", context.format("hello world"));
    }
}

File: src/TextFormatterTest.ts
import { TextFormatterStrategy } from './TextFormatterStrategy';
import { TextFormatterContext } from './TextFormatterContext';
import { UppercaseStrategy } from './UppercaseStrategy';
import { LowercaseStrategy } from './LowercaseStrategy';
import { TitleCaseStrategy } from './TitleCaseStrategy';

describe('TextFormatterStrategies', () => {
  test('UppercaseStrategy', () => {
    const uppercaseStrategy: TextFormatterStrategy = new UppercaseStrategy();
    expect(uppercaseStrategy.format('hello world')).toBe('HELLO WORLD');
    expect(uppercaseStrategy.format('Java Programming')).toBe('JAVA PROGRAMMING');
  });

  test('LowercaseStrategy', () => {
    const lowercaseStrategy: TextFormatterStrategy = new LowercaseStrategy();
    expect(lowercaseStrategy.format('HELLO WORLD')).toBe('hello world');
    expect(lowercaseStrategy.format('Java Programming')).toBe('java programming');
  });

  test('TitleCaseStrategy', () => {
    const titleCaseStrategy: TextFormatterStrategy = new TitleCaseStrategy();
    expect(titleCaseStrategy.format('hello world')).toBe('Hello World');
    expect(titleCaseStrategy.format('JAVA PROGRAMMING')).toBe('Java Programming');
  });

  test('TextFormatterContext', () => {
    const context = new TextFormatterContext(new UppercaseStrategy());
    expect(context.format('hello world')).toBe('HELLO WORLD');

    context.setStrategy(new LowercaseStrategy());
    expect(context.format('HELLO WORLD')).toBe('hello world');

    context.setStrategy(new TitleCaseStrategy());
    expect(context.format('hello world')).toBe('Hello World');
  });
});
